summerofcode/2023.md

```diff
@@ -129,6 +129,31 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
 - Difficulty: Hard
 - Upstream Issues (URL): https://github.com/kubevela/kubevela/issues/5366
 
+#### Auto-generate the TypeScript and Java languages API SDK
+
+- Description: Some developers start to integrate KubeVela into the enterprise system. it is increasingly important to provide API SDK to help developers start faster. Java and TypeScript are currently the two most in-demand languages. KubeVela API server follows the Open API schema, It means the Open API generator tools could provide some help.
+- Expected outcomes:
+  - Commit the high-quality SDK code, include the TypeScript and Java languages.
+  - Commit the GitHub action config and tools to auto-generate SDK.
+- Recommended Skills: Java, TypeScript, NodeJS, KubeVela, GithubAction
+- Mentor(s): QingGuo Zeng @barnettZQG
+- Expected project size: 175h
+- Difficulty: Medium
+- Upstream Issues (URL): https://github.com/kubevela/kubevela/issues/5428
+
+#### Expand multiple database drivers for the API server
+
+- Description: The API server of the KubeVela needs to save the metadata data to the database to obtain more efficient query efficiency and larger storage space. Now we only support MongoDB. Some users don't have the MongoDB infrastructure. Mysql and PostgreSQL drivers are needed by many users. Most users install the KubeVela without the database. When they want to use it for production, the existing data need to migrate to one database. The migration tool is important.
+- Expected outcomes:
+  - Commit the Mysql driver code.
+  - Commit the PostgreSQL driver code.
+  - Commit the migrate data tool code.
+- Recommended Skills: Go, KubeVela, Mysql, PostgreSQL
+- Mentor(s): QingGuo Zeng @barnettZQG
```

Comment:
@barnettZQG 
Can you add your email here and the idea above please?

Reply:
Done.